salt/modules/file.py

```diff
@@ -3781,10 +3781,24 @@ def is_hardlink(path):
     return res and res["st_nlink"] > 1
 
 
-def is_link(path):
+def is_link(path, nostat=False):
     """
     Check if the path is a symbolic link
 
+    Args:
+
+        path (str): The path to check if it is a link.
+
+        nostat (bool):
+            Use information from parent directory to determine if entry
+            is a symbolic link. This avoids the stat operation, which
+            may hang under certain circumstances.
```

Comment:
```suggestion
            Use information from parent directory to determine if entry
            is a symbolic link. This avoids the stat operation, which
            may hang under certain circumstances. For example, NFS mounts
            which have gone offline or are suffering some network issues.
            This will make the check quite slower on parent directories
            with a lot of files, but will reduce the chances of hanging.
```
If there's a better way to explain the scenario where this happened to you, and that is getting fixed, by all means, adjust the wording.